packages/iTwinUI-react/src/core/Modal/index.ts

```diff
@@ -4,8 +4,11 @@
  *--------------------------------------------------------------------------------------------*/
 export { Modal } from './Modal';
 export type { ModalProps } from './Modal';
-export { ModalButtonBar } from './ModalButtonBar';
-export type { ModalButtonBarProps } from './ModalButtonBar';
-export { ModalContent } from './ModalContent';
-export type { ModalContentProps } from './ModalContent';
+
+export { DialogButtonBar as ModalButtonBar } from '../Dialog/DialogButtonBar';
+export type { DialogButtonBarProps as ModalButtonBarProps } from '../Dialog/DialogButtonBar';
+
+export { DialogContent as ModalContent } from '../Dialog/DialogContent';
+export type { DialogContentProps as ModalContentProps } from '../Dialog/DialogContent';
```

Comment:
This is weird. Shouldn't we mark the old ones as deprecated?

Like this:
```tsx
/** @deprecated - Use `DialogButtonBar` instead. */
export const ModalButtonBar = DialogButtonBar;
```

Reply:
This PR is just for replacement, I plan to add deprecation tags with another PR where Dialog will be exposed.

Reply:
Then don't add deprecation but still `export const ModalButtonBar = DialogButtonBar` would be better. And I think you'd want to preserve the old prop types.

Reply:
What is the difference?

Reply:
Well, for props you need to bring back the old types anyway because the new ones use `ComponentPropsWithRef`

And for components, it's still better because It would make them distinct and allow using different jsdoc (so adding deprecation in the future is a single line change). And it's also less confusing for me as there is less "magic" happening in barrel file.

Reply:
Changed